### PR TITLE
[FLINK-9849] [hbase] Hbase upgrade to 2.0.1

### PR DESCRIPTION
**Pull request by zhangminglei**
## What is the purpose of the change

Upgrade hbase version to 2.0.1 for hbase connector

### CODE REVIEW

File: flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/TableInputFormat.java [hunk 1]
```diff
@@ -81,7 +85,9 @@ private HTable createTable() {
 		org.apache.hadoop.conf.Configuration hConf = HBaseConfiguration.create();
 
 		try {
-			return new HTable(hConf, getTableName());
+			Connection connection = ConnectionFactory.createConnection(hConf);
+			Table table = connection.getTable(TableName.valueOf(getTableName()));
+			return (HTable) table;
```
Comment (yanghua) [inline flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/TableInputFormat.java]: I think we should release the connection when happens exception

File: flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java [hunk 2]
```diff
@@ -87,22 +90,22 @@ public void configure(Configuration parameters) {
 
 		@Override
 		public void open(int taskNumber, int numTasks) throws IOException {
-			table = new HTable(conf, "flinkExample");
+			Connection connection = ConnectionFactory.createConnection(conf);
```
Comment (yanghua) [inline flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java]: I think we should add try / catch block to protect the connect leak.

Comment (yanghua) [inline flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java]: I think we should add try / catch block to protect the connect leak.

Comment (yanghua) [inline flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java]: I think we should add try / catch block to protect the connect leak.

Comment (yanghua) [reply flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java]: Based on [HBase Connection JavaDoc](https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/client/Connection.html#close--) it seems the caller should invoke `close` method to release resource? so I suggest we should close connection in udf's `close` method.

**Comment by zhangminglei** [comment]
Thanks @yanghua you are right! I just want waiting the travis ending. then give the old and new version's dependency tree.

**Comment by zhangminglei** [comment]
Hi, @tedyu I found an issue when upgrade the hbase version to either 2.0.1 or 2.1.0. Both can lead problem. ```Failed to read artifact descriptor for org.glassfish:javax.el:jar 3.0.1-b06-SNAPSHOT```. And then cause some problem. Maybe because of my network?

**Comment by zhangminglei** [comment]
@yanghua Could you please take on look this issue also ?

**Comment by yanghua** [comment]
seems `org.glassfish:javax.el:jar 3.0.1` has a non-SNAPSHOT version in maven repository, see here :  https://mvnrepository.com/artifact/org.glassfish/javax.el/3.0.1-b10, Do you use a inner maven repository hosted in your company? If yes, maybe it forbid you to download. You can exclude this dependency in hbase and introduce a single dependency in your pom about it.

**Comment by zhangminglei** [comment]
Hi, @yanghua I do not use a inner maven repo. It seems does not work for me.Exclude SNAPSHOT dependency and add a stable glassfish but still useless.

**Comment by yanghua** [comment]
hi @tedyu I talked with @zhangminglei it seems the new HBase version(2.1.0) depends on a SNAPSHOT version of `org.glassfish.javax.el`, however this SNAPSHOT version can not be found in public maven repository, can you check this? Thanks.

**Comment by zhangminglei** [comment]
Thanks @yanghua for pointing this out!

**Comment by greghogan** [comment]
Should this be a separate connector for HBase 2.x where we would also keep a connector for HBase 1.x?

**Comment by tedyu** [comment]
That would be ideal - a new connector for hbase 2.x considering that:

hbase 1.x is stable
hbase 2.y is newly released.

**Comment by tedyu** [comment]
```
16:08:53.680 [ERROR] src/main/java/org/apache/flink/addons/hbase/TableInputFormat.java:[31] (imports) ImportOrder: Import org.apache.hadoop.hbase.client.Result appears after other imports that it should precede
16:08:53.680 [ERROR] src/test/java/org/apache/flink/addons/hbase/HBaseTestingClusterAutostarter.java:[38] (imports) ImportOrder: Import org.apache.hadoop.hbase.client.HBaseAdmin appears after other imports that it should precede
```

**Comment by zentol** [comment]
Subsumed by FLINK-18795

**Pull request closed without merging**